tests/utils.go

```diff
@@ -241,6 +241,12 @@ type ObjectEventWatcher struct {
 	resourceVersion        string
 	startType              startType
 	dontFailOnMissingEvent bool
+	warningsIgnoreList     []string
+}
+
+type WarningsPolicy struct {
```

Comment:
There are 2 possible combination of `IgnoreWarnings` == false, where each one has different logic

1. IgnoreWarnings == false, len(WarningsIgnoreList)  == 0, means it starts the watcher, and assert + log on all warnings.
2. IgnoreWarnings == false, len(WarningsIgnoreList)  != 0, means it starts the watcher, and assert + log on all warnings beside the ignoreList ones.
So we see its isn't 1:1

I preferred to keep those 2 setters for ObjectEventWatcher, (so they are independent) unless you think its crucial.

wdyt ?

Reply:
Well I see the following if we try to normalize things 
three different things can happend withwarnings "ignore, log and log+failure" let's just do an enum with this three states and a list of warnings to ignore 

```golang
const (
	Ignore = iota
        Log
	LogAndFail
)

type WarningsPolicy struct {
    WarningAction int
    WarningSkipList []string
}
```

This wai we use only the policy around and we cover all the cases.

Reply:
as we spoke, failing to Ignore and LogAndFail (which will respect the WarningSkipList in case of LogAndFail)
without the Log, and reusing the struct at ObjectEventWatcher

Reply:
Done, as a separate commit, that i will squash at the end of the PR